room.h

```diff
@@ -0,0 +1,37 @@
+#ifndef ROOM_H
+#define ROOM_H
+
+#include <unordered_map>
+#include "sensor.h"
+#include "device.h"
+
+class Room
+{
+public:
+    Room(const std::string &name = "");
+
+    void AddSensor(Sensor &sensor);
+    void DeleteSensor(const std::string &name);
+    Sensor* GetSensor(const std::string &name);
+
+    void AddDevice(Device &device);
+    void DeleteDevice(const std::string &name);
+    Device* GetDevice(const std::string &name);
+
+    std::string GetName();
+    void SetName(const std::string &name);
+
+    std::unordered_map<std::string, Sensor> GetSensors();
+    std::unordered_map<std::string, Device> GetDevices();
+
+
+private:
+    std::string name_;
+    std::unordered_map<std::string, Sensor> sensors_;
+    std::unordered_map<std::string, Device> devices_;
```

Comment:
my problem with the unordered_map in your case is that you already keep the name in each sensor.name field, you don't need to keep it in the sensors_->first as well, I don't see any justification for keeping the same data twice. You could have a vector or list and search by the name just as easily.

Reply:
Same for device.

Reply:
It make sense, I didn't saw things from that perspective, I will update my code